cupy/testing/random.py

```diff
@@ -0,0 +1,60 @@
+import atexit
+import numpy
+
+import cupy
+
+
+# In some tests (which utilize condition.repeat or condition.retry),
+# setUp/tearDown is nested. setup_random() and teardown_random() do their
+# work only in the outermost setUp/tearDown pair.
+_nest_count = 0
+
+
+@atexit.register
+def _check_teardown():
+    assert _nest_count == 0, ('setup_random() and teardown_random() '
+                              'must be called in pairs.')
+
+
+def setup_random(numpy_seed=None, cupy_seed=None):
+    """Sets up the deterministic random states of ``numpy`` and ``cupy``.
+
+    Args:
+         numpy_seed(int or None): Seed to initialize ``numpy.random``. If
+         ``None``, predetermined constant value is used.
+         cupy_seed(int or None): Seed to initialize ``cupy.random``. If
+         ``None``, predetermined constant value is used.
+    """
+    global _nest_count
+    if _nest_count == 0:
+        numpy.random.seed(100 if numpy_seed is None else numpy_seed)
+        cupy.random.seed(101 if cupy_seed is None else cupy_seed)
+    _nest_count += 1
+
+
+def teardown_random():
+    """Tears down the deterministic random states set up by ``setup_random``.
+
+    """
+    global _nest_count
+    assert _nest_count > 0
+    _nest_count -= 1
+    if _nest_count == 0:
+        numpy.random.seed(None)
+        cupy.random.reset_states()
+
+
+def get_random_state(xp, seed=None):
```

Comment:
No, it's not used.

Actually, `get_random_state` function itself is not used for now.
The purpose of this function is to fix the random parameters used in `product` or `parameterize` (e.g. https://github.com/cupy/cupy/blob/master/tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py#L55).
Currently, all the tests using such random parameters are not probabilistic (parameters are random but the tests are always successful).
So I didn't use `get_random_state` and left it for future use.